src/main/java/com/minecolonies/coremod/colony/buildings/workerbuildings/BuildingBarracks.java

```diff
@@ -124,14 +124,14 @@ public void registerBlockPosition(@NotNull final BlockState block, @NotNull fina
         {
             if (world.getBlockState(pos).getBlock() != ModBlocks.blockHutBarracksTower)
             {
-                world.setBlockState(pos, ModBlocks.blockHutBarracksTower.getDefaultState().with(BlockBarracksTowerSubstitution.FACING, block.get(AbstractBlockHut.FACING)));
-                final TileEntity tile = world.getTileEntity(pos);
+                world.setBlockAndUpdate(pos, ModBlocks.blockHutBarracksTower.defaultBlockState().setValue(BlockBarracksTowerSubstitution.FACING, block.getValue(AbstractBlockHut.FACING)));
```

Comment:
While it doesn't actually _matter_, since `BlockBarracksTowerSubstitution.FACING` and `AbstractBlockHut.FACING` are equal references, technically this logic is backwards and these two symbols should be reversed.

Reply:
agreed, will adjust this in this PR anyway, while it doesn't really matter, it's small enough to adjust in this PR